### ISSUE TITLE
Compilation fail on FreeBSD 13.1

### ISSUE THREAD

**Issue by xjjx** [spdk/spdk#2567]
```
in /git_repos/spdk_repo/spdk/test/common/config/pkgdep/git:100 -> install_refspdk(["--install-deps"],["--upgrade"],["--disable-tsocks"])
     ...
   95                   if [[ $OSID != freebsd ]]; then
   96                           $MAKE -C $output_dir $MAKEFLAGS include/spdk/config.h
   97                           CONFIG_OCF_PATH="$output_dir/ocf" $MAKE -C $output_dir/lib/env_ocf $MAKEFLAGS exportlib O=$output_dir/ocf.a
   98                           $output_dir/configure $config_params --with-ocf=$output_dir/ocf.a --with-shared --without-fio
   99                   fi
=> 100                  $MAKE -C $output_dir $MAKEFLAGS
   101          )
   102  }
   103
   104  function install_qat() {
   105          # Dissect the kernel version into maj, min, release and local version
     ...

---------

[84/240] Compiling C object lib/librte_eal.a.p/eal_common_eal_common_options.c.o
FAILED: lib/librte_eal.a.p/eal_common_eal_common_options.c.o
cc -Ilib/librte_eal.a.p -Ilib -I../lib -I. -I.. -Iconfig -I../config -Ilib/eal/include -I../lib/eal/include -Ilib/eal/freebsd/include -I../lib/eal/freebsd/include -Ilib/eal/x86/include -I../lib/eal/x86/include -Ilib/eal/common -I../lib/eal/common -Ilib/eal -I../lib/eal -Ilib/kvargs -I../lib/kvargs -I../lib/metrics -Ilib/telemetry -I../lib/telemetry -fcolor-diagnostics -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -O0 -g -include rte_config.h -Wcast-qual -Wdeprecated -Wformat -Wformat-nonliteral -Wformat-security -Wmissing-declarations -Wmissing-prototypes -Wnested-externs -Wold-style-definition -Wpointer-arith -Wsign-compare -Wstrict-prototypes -Wundef -Wwrite-strings -Wno-address-of-packed-member -Wno-missing-field-initializers -D_GNU_SOURCE -D__BSD_VISIBLE -fPIC -Wno-error -fPIC -march=native -DALLOW_EXPERIMENTAL_API -DALLOW_INTERNAL_API '-DABI_VERSION="22.1"' -DRTE_LIBEAL_USE_GETENTROPY -DRTE_LOG_DEFAULT_LOGTYPE=lib.eal -MD -MQ lib/librte_eal.a.p/eal_common_eal_common_options.c.o -MF lib/librte_eal.a.p/eal_common_eal_common_options.c.o.d -o lib/librte_eal.a.p/eal_common_eal_common_options.c.o -c ../lib/eal/common/eal_common_options.c
../lib/eal/common/eal_common_options.c:1970:3: error: too few arguments provided to function-like macro invocation
                RTE_CPU_OR(cpuset, cpuset, &lcore_config[lcore_id].cpuset);
                ^
../lib/eal/freebsd/include/rte_os.h:42:19: note: expanded from macro 'RTE_CPU_OR'
        CPU_OR(&tmp, src2); \
                         ^
/usr/include/sys/cpuset.h:60:9: note: macro 'CPU_OR' defined here
#define CPU_OR(d, s1, s2)               __BIT_OR2(CPU_SETSIZE, d, s1, s2)
        ^
../lib/eal/common/eal_common_options.c:1970:3: error: use of undeclared identifier 'CPU_OR'
                RTE_CPU_OR(cpuset, cpuset, &lcore_config[lcore_id].cpuset);
                ^
../lib/eal/freebsd/include/rte_os.h:42:2: note: expanded from macro 'RTE_CPU_OR'
        CPU_OR(&tmp, src2); \
        ^
../lib/eal/common/eal_common_options.c:1972:2: error: too few arguments provided to function-like macro invocation
        RTE_CPU_NOT(cpuset, cpuset);
        ^
../lib/eal/freebsd/include/rte_os.h:61:22: note: expanded from macro 'RTE_CPU_NOT'
        CPU_ANDNOT(&tmp, src); \
                            ^
/usr/include/sys/cpuset.h:62:9: note: macro 'CPU_ANDNOT' defined here
#define CPU_ANDNOT(d, s1, s2)           __BIT_ANDNOT2(CPU_SETSIZE, d, s1, s2)
        ^
../lib/eal/common/eal_common_options.c:1972:2: error: use of undeclared identifier 'CPU_ANDNOT'
        RTE_CPU_NOT(cpuset, cpuset);
        ^
../lib/eal/freebsd/include/rte_os.h:61:2: note: expanded from macro 'RTE_CPU_NOT'
        CPU_ANDNOT(&tmp, src); \
        ^
../lib/eal/common/eal_common_options.c:1978:2: error: too few arguments provided to function-like macro invocation
        RTE_CPU_AND(cpuset, cpuset, &default_set);
        ^
../lib/eal/freebsd/include/rte_os.h:35:20: note: expanded from macro 'RTE_CPU_AND'
        CPU_AND(&tmp, src2); \
                          ^
/usr/include/sys/cpuset.h:61:9: note: macro 'CPU_AND' defined here
#define CPU_AND(d, s1, s2)              __BIT_AND2(CPU_SETSIZE, d, s1, s2)
        ^
../lib/eal/common/eal_common_options.c:1978:2: error: use of undeclared identifier 'CPU_AND'
        RTE_CPU_AND(cpuset, cpuset, &default_set);
        ^
../lib/eal/freebsd/include/rte_os.h:35:2: note: expanded from macro 'RTE_CPU_AND'
        CPU_AND(&tmp, src2); \
        ^
6 errors generated.
```

**Comment by mikeBashStuff** [comment]
I believe this one was already reported.

**Comment by mikeBashStuff** [comment]
Sorry, not reported, but I mentioned the very same issue under: https://github.com/spdk/spdk/issues/2542#issuecomment-1148552119

**Comment by jimharris** [comment]
I've cherry-picked the DPDK fix onto our spdk-22.03 branch:

https://review.spdk.io/gerrit/c/spdk/dpdk/+/13468
